#sonar params
usesonar: false
a00: 0.0
a01: 0.5
a10: 0.3
a11: 0.9
a20: 0.8
a21: 0.6
a30: 0.8
a31: 0.6
asonar: 2.0

a_theta: 0.99
b_alpha: 0.1
c_dist: 0.125

max_vx: 0.5
max_rz: 0.87
min_turn_radius: 0.0
#1.25

#robot's width
robot_width: 0.3

# detection tolarent of obstacles, the smaller the stricter
det_tolar: 3

# max distance error between two edges of safezone
safezone_rest_wide: 10.8

# max direction error between goal and nearest safe zone
largest_err_direction: 1.57

# min range error of a safezone = min_err_range_sz * range_step_size (m)
min_err_range_sz: 3

# width threshold(m) for safezone detection = lvlerrthreshold_of_sz * range_step_size, least 1
lvlerrthreshold_of_sz: 1

# absolute safe distance = lvlabsulute_safe * range_step_size
lvlabsulute_safe: 0.5

#half field of view(rad), 0.52rad=30degree
half_fov: 1.52

#lidar
uselidar: true
viewsafezonelidar: true
# detect range(m) = range_step_size(m) * obs_timesofdeld, both 3d and 2d lidar
obs_timesofdeld: 25
# step size of each step in range in one direction, 2d
range_step_size: 0.1
# obs detect increment angle(rad) around z axix, both 3d and 2d lidar. The same to angle_increment.
obs_dangle: 0.005
# diameter of search circle diameter, both 3d and 2d lidar
deld: 0.6
# max turn speed default pi/2
OA_MAX_ANGULAR_SPEED: 1.5708
# tell program that data from -time_of_halfpi*pi/2 to time_of_halfpi*pi/2 is ok
# for 2d and 3d lidar
time_of_halfpi: 1.5

#whether used in visual navigation. Get param from **.yaml
used_in_visual_obsavoid: true 

#3d lidar
use3dlidar: false
viewsafezone3dlidar: false
# start detection height of 3dlidar frame
sheight_of_3dlidar: 0.0
# lidar's translation from base_link(m), for both 3d and 2d lidar
lx: 0.25
ly: 0
lz: 0
#num of scan layer
detect_layer: 1
#dead zone, radius of circle around robot you want to ignore
deadzone: 0.0

#tof
usetof: false
viewsafezonetof: false
# detect range(m) = deld(m) * timesofdeld
obs_timesofdeld_tof: 5
# obs detect increment angle(rad)
obs_dangle_tof: 0.035
# search circle diameter
deld_tof: 0.46
# start detection height of camera frame
sheight_of_cam: -0.42

# datatype for viewer 3d
#1-tof,rgbd
#2-3d lidar
data_type: 2
